copycat-api/src/main/java/org/apache/kafka/copycat/storage/Converter.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.storage;
+
+/**
+ * The Converter interface provides support for translating between Copycat's runtime data format
+ * and the "native" runtime format used by the serialization layer. This is used to translate
+ * two types of data: records and offsets. The (de)serialization is performed by a separate
```

Comment:
HBase has the CheckAndPut API that seem like could do dedup if we provide the correct thing to "check".